src/main/resources/liquibase/crop_changelog/18_3_0.xml

```diff
@@ -104,4 +104,397 @@
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="1">
+				SELECT
+				COUNT(*)
+				FROM
+				information_schema.routines
+				WHERE
+				routine_schema = DATABASE()
+				AND specific_name = 'migrate_attr_udflds_to_ontology';
+			</sqlCheck>
+		</preConditions>
+		<comment>Run migrate_attr_udflds_to_ontology procedure</comment>
+		<sql dbms="mysql" splitStatements="false">
+			call migrate_attr_udflds_to_ontology();
+		</sql>
+	</changeSet>
+
+	<changeSet author="clarysabel" id="v18.3.0-16">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="1">
+				SELECT
+				COUNT(*)
+				FROM
+				information_schema.routines
+				WHERE
+				routine_schema = DATABASE()
+				AND specific_name = 'migrate_attr_udflds_to_ontology';
+			</sqlCheck>
+		</preConditions>
+		<sql dbms="mysql" splitStatements="false">
+			DROP PROCEDURE migrate_attr_udflds_to_ontology;
+		</sql>
+	</changeSet>
+
+	<changeSet author="clarysabel" id="v18.3.0-17">
+		<preConditions onFail="MARK_RAN">
+			<and>
+				<tableExists tableName="udflds_attributes_bkp"></tableExists>
+				<not>
+					<tableExists tableName="atributs_invalid_atype_bkp"></tableExists>
+				</not>
+				<sqlCheck expectedResult="1">
+					SELECT COUNT(*) > 0
+					FROM udflds_attributes_bkp
+					WHERE
+					cvterm_id is not null;
+				</sqlCheck>
+				<not>
+					<foreignKeyConstraintExists foreignKeyName="fk_atributs_atype_cvterm"></foreignKeyConstraintExists>
+				</not>
+			</and>
+		</preConditions>
+		<comment>Connect attributes with the new ontology variables</comment>
+		<sql dbms="mysql">
+				ALTER TABLE atributs
+				  CHANGE atype old_atype_bkp int(11);
+
+				ALTER TABLE atributs
+				  ADD COLUMN atype int(11) DEFAULT NULL AFTER gid;
+
+				UPDATE atributs INNER JOIN udflds_attributes_bkp bkp ON bkp.fldno = old_atype_bkp
+				SET atype = bkp.cvterm_id WHERE 1 = 1;
+
+				CREATE TABLE atributs_invalid_atype_bkp AS
+				SELECT * FROM atributs WHERE atributs.atype IS NULL;
+				DELETE FROM atributs WHERE atributs.atype IS NULL;
+
+				ALTER TABLE atributs DROP COLUMN old_atype_bkp;
+				ALTER TABLE atributs MODIFY atype int(11) NOT NULL;
+
+				ALTER TABLE atributs
+				ADD CONSTRAINT fk_atributs_atype_cvterm FOREIGN KEY (atype) REFERENCES cvterm(cvterm_id);
+		</sql>
+	</changeSet>
+
+	<changeSet author="clarysabel" id="v18.3.0-18">
+		<preConditions onFail="MARK_RAN">
+			<and>
+				<tableExists tableName="udflds_attributes_bkp"></tableExists>
+				<sqlCheck expectedResult="1">
+					SELECT COUNT(*) > 0
+					FROM udflds_attributes_bkp
+					WHERE
+					cvterm_id is not null;
+				</sqlCheck>
+				<sqlCheck expectedResult="1">
+					select count(*) > 0 from listdataprops l inner join udflds_attributes_bkp bkp on bkp.fcode = l.column_name;
+				</sqlCheck>
+			</and>
+		</preConditions>
+		<comment>Replace old attributes names in listdataprops with the new variable names</comment>
+		<sql dbms="mysql">
+			ALTER TABLE listdataprops
```

Comment:
Why did we have to add a temporary column: `column_name_tmp`? Can't we just do a direct update on existing column_name?

UPDATE listdataprops INNER JOIN udflds_attributes_bkp bkp ON bkp.fcode = column_name
			SET column_name = UPPER(bkp.cvterm_name) WHERE 1 = 1;

Reply:
We can do it here and in attributs table but the total execution time was higher so it performs much better to have a temporary column in both cases